src/plugins/discover/public/application/angular/discover.js

```diff
@@ -208,6 +219,11 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
 
   const history = getHistory();
 
+  let initialSearchSessionId = $routeParams[SEARCH_SESSION_ID_QUERY_PARAM];
+  if (initialSearchSessionId) {
+    removeQueryParam(history, SEARCH_SESSION_ID_QUERY_PARAM);
```

Comment:
Hmm, this behavior seems to work differently than the dashboard one. If I set the param in the URL in the dashboard, it would stay in the URL until I changed the query somehow. In this version, it gets removed right away rather than when I make changes to the query. Which behavior is desirable? It still seems to get sent to `searchSource` regardless, but I think we should have the same behavior in both.

Reply:
Thanks for noticing. I think dashboard behavior is better. Will try to achieve the same here

Reply:
Improved. I had to re-introduced same query param helper for this as in https://github.com/elastic/kibana/pull/81489/files#diff-162a8c20c8b48e47661f530ae56e74522effa428d1be410569cbcf09b4e6a6b3
to get the latest query state from history (this isn't propagated to angular $routeParams)